apps/firehose_to_es_processor/lib/firehose_record.py

```diff
@@ -40,4 +49,29 @@ def _transform_and_extract_from_log_event(self, log_event):
             for k, v in transformed_message.items():
                 transformed_payload[k] = json.dumps(v)
 
+        message = transformed_payload['@message']
+        log_group = transformed_payload['@log_group']
+        log_stream = self.record["logStream"]
+        if airbrake_notifier and self._is_message_appropriate_for_airbrake(message, log_group):
+            airbrake_error = "'{0} {1} '@log_stream': {2}".format(log_group, message, log_stream)
+            try:
+                airbrake_notifier.notify(str(airbrake_error))
+            except:
+                pass
+
         return transformed_payload
+
+    def _is_message_appropriate_for_airbrake(self, message, log_group):
+        send_to_airbrake = False
+
+        blacklisted_log_group_names = os.environ.get("AIRBRAKE_BLACKLISTED_LOG_GROUP_NAMES")
+        blacklisted_log_group_names_regex_string = "|".join(blacklisted_log_group_names.split())
+        blacklisted_log_group_names_regexp = re.compile(blacklisted_log_group_names_regex_string, re.IGNORECASE)
+
+        whitelisted_log_message_terms = os.environ.get("AIRBRAKE_WHITELISTED_LOG_MESSAGE_TERMS")
```

Comment:
I'm pretty sure that calling `os.environ.get` each time this method runs will 1) make a system call to get from the environment and 2) recompile the regex every time this function runs. Best practice is to make this static or global. Probably best to be static in this case. See https://docs.aws.amazon.com/lambda/latest/dg/best-practices.html